services/utilServices/AuthService.js

```diff
@@ -22,10 +25,12 @@ class AuthService {
   }
 
   async getGithubAuthToken({ csrfState, code, state }) {
-    const isCsrfValid =
-      jwtUtils.verifyToken(csrfState) &&
-      jwtUtils.decodeToken(csrfState).state === state
-    if (!isCsrfValid) throw new ForbiddenError()
+    try {
+      const decoded = jwtUtils.verifyToken(csrfState)
+      if (decoded.state !== state) throw new Error("State does not match")
+    } catch (err) {
+      throw new ForbiddenError()
```

Comment:
Makes sense, 668a689

Reply:
actually i was thinking of setting it on the error so that the user gets a better diagnostic @__@ It'll be rather tedious if we don't have a custom error though, we'll have to resort to something like the `notLoggedIn` error above, where we manually set `error.name` then use `if/else` in the `catch` branch

if we are to log the error, it's for internal consumption and hence, should include information that could help us in troubleshooting (eg: `csrfstate`/`code`/`state`)

Reply:
3f8e038